CCallback.cpp

```diff
@@ -217,6 +217,16 @@ bool CCallback::buildBuilding(const CGTownInstance *town, BuildingID buildingID)
 	return true;
 }
 
+bool CCallback::triggerTownSpecialBuildingAction(const CGTownInstance *town, BuildingSubID::EBuildingSubID subBuildingID)
+{
+	if(town->tempOwner!=player)
+		return false;
+
```

Comment:
Should probably be handled in UI?

Reply:
adapted this logic from `CCallback::buildBuilding`